[EDU-3600] feat: add gql dataset bot (#564)
aziontech/docs: src/content/docs/en/pages/devtools/api-graphql/features/metrics-fields/metrics-fields.mdx

| streamedLines | Total amount of lines streamed to the configured endpoint. Maximum value of `2000`. This field is the result of a sum. Example: `837` |
| ts | Timestamp of when the event was created. Example: `2022-10-20T10:10:10` |

---

## botProtectionMetrics (Bot Protection)

| Field | Description |
| ----- | ----------- |
| botMode | The environment the Bot Protection function is running on. Can be `web` or `api`.  |

> | classified |  |

Author: ```suggestion
| classified | Classification of different types of bots and traffic. Can be a legitimate traffic, a good bot, or a bat bot. |
```

> | clientId |  |

Author: ```suggestion
| clientId | Unique identifier of the client. |
```

> | countryVariation | Country variation by a fingerprint. |

Author: ```suggestion
| countryVariation | Country variation from a specific fingerprint. |
```

> | customFieldVariation | Fields variations by a fingerprint. |

Author: ```suggestion
| customFieldVariation | Fields variations from a specific fingerprint. |
```

> | errorCount |  |
> | errorRate | Error by a fingerprint. |

Author: ```suggestion
| errorCount | Total count of error from a specific fingerprint. |
| errorRate | Rate of error from a specific fingerprint. |
```

> | fingerprint |  |

Author: ```suggestion
| fingerprint | Unique hash for devices accessing your edge applications. |
```

| frequencyRate | Rate of frequency a fingerprint accessed an application. |

> | headRequestCount |  |

Author: ```suggestion
| headRequestCount | Total count of HEAD requests from a specific fingerprint. |
```

> | headRequestFrequencyRate |  |

Author: ```suggestion
| headRequestFrequencyRate | Rate of frequency of HEAD requests from a specific fingerprint.  |
```

| host | Host information sent on the request line. Stores: host name from the request line, or host name from the “Host” request header field, or the server name matching a request. Example: `g1sdetynmxe0ao.map.azionedge.net` |
| logTag |  |

> | postRequestCount | POST requests by a fingerprint. |

Author: ```suggestion
| postRequestCount | Total count of POST requests from a specific fingerprint. |
```

> | postRequestFrequencyRate |  |

Author: ```suggestion
| postRequestFrequencyRate | Rate of frequency of POST requests from a specific fingerprint. |
```

| remoteAddr | IP address of the origin that generated the request. Example: `127.0.0.1` |

> | requestCount |  |

Author: ```suggestion
| requestCount | Total count of requests made to your edge application. |
```

> | requestUriVariation | Variation of a unique URI. |

Author: ```suggestion
| requestUriVariation | Variation of requests with a unique URI. |
```

| ts | Timestamp of when the event was created. Example: `2022-10-20T10:10:10` |

> | uniqueCountryCount | Count of country variation by a fingerprint. |
> | uniqueCustomFieldCount | Count of custom fields by a fingerprint. |
> | uniqueRemoteAddressCount |  |
> | uniqueRequestUriCount | Unique URIs by a fingerprint. |

Author: ```suggestion
| uniqueCountryCount | Count of unique country variation from a specific fingerprint. |
| uniqueCustomFieldCount | Count of custom fields from a specific fingerprint. |
| uniqueRemoteAddressCount | Count of unique IP addresses from a specific fingerprint. |
| uniqueRequestUriCount | Count of unique URIs from a specific fingerprint. |
```

### Calculated fields

When a field is a result of any kind of calculation, such as a sum, they're considered as *calculated fields*.

| Calculated field | Description |
| ---------------- | ----------- |

> | countryVariationAvg | Average of country variation by a fingerprint. |

Author: ```suggestion
| countryVariationAvg | Average of country variation from a specific fingerprint. |
```

> | countryVariationMedian |  |

Author: ```suggestion
| countryVariationMedian | Median of country variation from a specific fingerprint. |
```

> | customFieldVariationAvg | Average of fields variations by a fingerprint. |
> | customFieldVariationMedian |  |
> | errorCountAvg |  |
> | errorCountMedian |  |
> | errorRateAvg | Average of error by a fingerprint. |
> | errorRateMedian |  |
> | frequencyRateAvg | Average of the rate of frequency a fingerprint accessed an application. |
> | frequencyRateMedian |  |
> | headRequestCountAvg |  |
> | headRequestCountMedian |  |
> | headRequestFrequencyRateAvg |  |
> | headRequestFrequencyRateMedian |  |
> | postRequestCountAvg | Average of POST requests by a fingerprint. |
> | postRequestCountMedian |  |
> | postRequestFrequencyRateAvg |  |
> | postRequestFrequencyRateMedian |  |
> | requestCountAvg |  |
> | requestCountMedian |  |
> | requestUriVariationAvg | Average of variation of a unique URI. |
> | requestUriVariationMedian |  |
> | uniqueCountryCountAvg | Average of the count of country variation by a fingerprint. |
> | uniqueCountryCountMedian |  |
> | uniqueCustomFieldCountAvg | Average of the count of custom fields by a fingerprint. |
> | uniqueCustomFieldCountMedian |  |
> | uniqueRemoteAddressCountAvg |  |
> | uniqueRemoteAddressCountMedian |  |
> | uniqueRequestUriCountAvg | Average of unique URIs by a fingerprint. |
> | uniqueRequestUriCountMedian |  |

Author: ```suggestion
| customFieldVariationAvg | Average of fields variations from a specific fingerprint. |
| customFieldVariationMedian | Median of fields variations from a specific fingerprint. |
| errorCountAvg | Average of count of error from a specific fingerprint. |
| errorCountMedian | Median of count of error from a specific fingerprint. |
| errorRateAvg | Average of rate of error from a specific fingerprint. |
| errorRateMedian | Median of rate of error from a specific fingerprint.  |
| frequencyRateAvg | Average of the rate of frequency a fingerprint accessed an application. |
| frequencyRateMedian | Median of the rate of frequency a fingerprint accessed an application. |
| headRequestCountAvg | Average of count of HEAD requests from a specific fingerprint. |
| headRequestCountMedian | Median of count of HEAD requests from a specific fingerprint. |
| headRequestFrequencyRateAvg | Average of rate of frequency of HEAD requests from a specific fingerprint. |
| headRequestFrequencyRateMedian | Median of rate of frequency of HEAD requests from a specific fingerprint. |
| postRequestCountAvg | Average of count of POST requests from a specific fingerprint. |
| postRequestCountMedian | Median of count of POST requests from a specific fingerprint. |
| postRequestFrequencyRateAvg | Average of rate of frequency of POST requests from a specific fingerprint. |
| postRequestFrequencyRateMedian | Median of rate of frequency of POST requests from a specific fingerprint. |
| requestCountAvg | Average of count of requests made to your edge application. |
| requestCountMedian | Median of count of requests made to your edge application. |
| requestUriVariationAvg | Average of variation of requests with a unique URI. |
| requestUriVariationMedian | Median of variation of requests with a unique URI. |
| uniqueCountryCountAvg | Average of count of unique country variation from a specific fingerprint. |
| uniqueCountryCountMedian | Median of count of unique country variation from a specific fingerprint. |
| uniqueCustomFieldCountAvg | Average of count of custom fields from a specific fingerprint. |
| uniqueCustomFieldCountMedian | Median of count of custom fields from a specific fingerprint. |
| uniqueRemoteAddressCountAvg | Average of count of unique IP addresses from a specific fingerprint. |
| uniqueRemoteAddressCountMedian | Median of count of unique IP addresses from a specific fingerprint. |
| uniqueRequestUriCountAvg | Average of count of unique URIs from a specific fingerprint. |
| uniqueRequestUriCountMedian | Median of count of unique URIs from a specific fingerprint. |
```
